modcc/cudaprinter.cpp

```diff
@@ -164,6 +167,36 @@ CUDAPrinter::CUDAPrinter(Module &m, bool o)
             }
         }
     }
+
+    // print the write_back kernel
+    if(module_->write_backs().size()) {
+        buffer().add_line("__global__");
+        buffer().add_line("void write_back_"+module_name_+"("+pack_name()+" params_) {");
+        buffer().increase_indentation();
+        buffer().add_line("using value_type = arb::fvm_value_type;");
+
+        buffer().add_line("auto tid_ = threadIdx.x + blockDim.x*blockIdx.x;");
+        buffer().add_line("auto const n_ = params_.n_;");
+        buffer().add_line("if(tid_<n_) {");
+        buffer().increase_indentation();
+
+        auto ion_prefix = [](ionKind k) {
+            return k==ionKind::Ca? "ca": k==ionKind::Na? "na": "k";
+        };
```

Comment:
The mappings between ion kind and names occur in a number of places in the code now; can we bring them together in the one spot? (This will also ease the transition to general ions).

Reply:
I have consoldiated this a bit, but I haven't done an exhaustive search.
This is going to get ripped up when generic ion species are introduced, so I am not inclined to really refactor fully now.